pkg/clients/ecr/repository_policy.go

```diff
@@ -0,0 +1,236 @@
+package ecr
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/crossplane/provider-aws/apis/ecr/v1alpha1"
+)
+
+const (
+	// RepositoryPolicyNotFoundException policy was not found
+	RepositoryPolicyNotFoundException = "RepositoryPolicyNotFoundException"
+
+	errNotSpecified = "failed to format Repository Policy, no rawPolicy or policy specified"
+)
+
+// RepositoryPolicyClient is the external client used for Repository Policy Resource
+type RepositoryPolicyClient interface {
+	SetRepositoryPolicyRequest(input *ecr.SetRepositoryPolicyInput) ecr.SetRepositoryPolicyRequest
+	DeleteRepositoryPolicyRequest(input *ecr.DeleteRepositoryPolicyInput) ecr.DeleteRepositoryPolicyRequest
+	GetRepositoryPolicyRequest(input *ecr.GetRepositoryPolicyInput) ecr.GetRepositoryPolicyRequest
+}
+
+// GenerateSetRepositoryPolicyInput Generates the CreateRepositoryInput from the RepositoryPolicyParameters
+func GenerateSetRepositoryPolicyInput(params *v1alpha1.RepositoryPolicyParameters, policy *string) *ecr.SetRepositoryPolicyInput {
+	c := &ecr.SetRepositoryPolicyInput{
+		RepositoryName: params.RepositoryName,
+		RegistryId:     params.RegistryID,
+		PolicyText:     policy,
+		Force:          params.Force,
+	}
+
+	return c
+}
+
+// LateInitializeRepositoryPolicy fills the empty fields in *v1alpha1.RepositoryPolicyParameters with
+// the values seen in ecr.GetRepositoryPolicyResponse.
+func LateInitializeRepositoryPolicy(in *v1alpha1.RepositoryPolicyParameters, r *ecr.GetRepositoryPolicyResponse) { // nolint:gocyclo
+	if r == nil {
+		return
+	}
+	in.RegistryID = r.RegistryId
```

Comment:
```suggestion
	in.RegistryID = awsclient.LateInitializeStringPtr(in.RegistryID, r.RegistryId)
```

We can't override spec if it already exists.